### ISSUE TITLE
Blank Screen in OBS Studio 25.04 on Win 10

### ISSUE THREAD

**Issue by timboxyz** [steveseguin/obsninja#65]
Start a stream on Android phone and it views fine in Chrome on Win 10. However, in OBS the view remains transparent. I have tried the various tick box options but no change in behaviour.

I note that in Chrome the screen is blank until I click on the play icon bottom left. Is it this that is preventing OBS from showing anything?

Interestingly enough on OSX 25.06 (which I know is not recommended), I get the permaid invite page all the time.

OBS installed from Win/OSX installers from OBS website.

**Comment by timboxyz** [comment]
> interestingly enough on OSX 25.06 (which I know is not recommended), I get the permaid invite page all the time

Although on a retry the preview window shows a black screen with the play button etc at the bottom of the screen as if it is waiting for the button to be pressed.

**Comment by steveseguin** [comment]
I'm a bit confused as I haven't heard of this issue so far.

You should not get the Invite page if the stream exists, but unable to load.

If you get a spinning window loading message, that is normally the client is unable to play the video or there is a network issue.

If it works in Chrome, but not OBS, perhaps it's a browser caching issue?

**Comment by steveseguin** [comment]
maybe try the beta version obs.ninja/beta to see if it works?

**Comment by timboxyz** [comment]
Ah, there seems to be a more fundamental problem with OBS Studio 25.. on Win 10, basically, the browser source is not working at all! I have tried both a full install and  the Portable Apps version and they both fail to work with a browser source just using the test URL. :(

I need to investigate this further.

BTW 25.0.7 is now out for OSX, should this version now work?

**Comment by steveseguin** [comment]
I'm rather confused by the problems you are running into, as you are perhaps the first to have these issues.

Sadly, the OBS devs were unable to fix the issue and so it remains in version 25 of OBS for Mac. It's not that they don't want to fix it, but the Chrome (CEF) team have not fully supported MacOS in their code yet, which is blocking the OBS team, which is in turn blocking me.   

I can send you a test link from my computer, and maybe it will work better?  It might also let me debug the issue.

**Comment by steveseguin** [comment]
I think I may have found someone else with this issue, so I'll try to debug this further

**Comment by timboxyz** [comment]
> Ah, there seems to be a more fundamental problem with OBS Studio 25.. on Win 10
Doh! Its that wretched browser hardware acceleration bug. Why it defaults to enabled when it's known not to work is beyond me....

I only ran on OSX for comparison, I wasn't expecting it to/need it to work properly.
What I now consistently get on that OS is the page loading as far as seeing the play controls but then waiting to load the video stream with the constant spinning icon. Same on the beta version if that is any help to you.

**Comment by b00rek** [comment]
I think we're talking about multiple issues here. I can't comment on hardware acceleration related one.

I'm using Win 10, 1803 build and OBS 25.0.4. I can reproduce the transparent screen issue. However, it isn't OBS-related in my case. As you already figured out @timboxyz, it has to do with having to press the play button in video player sometimes and it happens in Chrome too. It's rare, but if it happens that autoplay doesn't work, I just click on `Refresh browser cache of current page` in OBS properties of the web source, it reloads the page, and it works.

**Comment by steveseguin** [comment]
Auto-play issues in Chrome is common, as Chrome does not allow videos to auto play until the window has had "user interaction".

In the newest beta version, I've made it so the 'control-bar' is hidden, and so there will be no play button.  If you wish to play a video, you just need to click the screen and it should auto-play that video.

In OBS, videos should auto-play though. If it's a spinning grey screen, and you click the video in OBS via "interact", and it plays then, then there is something wrong with my code. If it still does not auto play, then that is likely a network issue or OBS v24/25 issue with Macs.

If you get "disconnected" from Websockets, in OBS browser source, you will need to refresh OBS Browser Source to reconnect.  If not, it won't be aware that the video has become to stream. 

I need to add in better handling of errors. I'm a bit concerned of course about throwing errors up inside OBS, as I do not know if OBS is live on air or not.  

Perhaps I can have OBS "auto reload" if an error occurs, with a 5-second timeout or something.

**Comment by b00rek** [comment]
> I'm a bit concerned of course about throwing errors up inside OBS, as I do not know if OBS is live on air or not.

Very good point. I have actually slightly modified the html file, at the point where it checks if "streamid" parameter is present in the URL, I've made it a bit more "production ready" so it shows only the spinning wheel and after 2 seconds shows a "no video" icon. I hacked it in a really ugly way, but with a bit more time, I'll try to add checking if URL has parameter like "production", and if it does, it could show "pretty errors", if it doesn't, it could show normal screen with more information.
